### PR TITLE
bump to java 16 so we can build 1.17

### PR DESCRIPTION
**Pull request by brandonmcclure**
remove UseCGroupMemoryLimitForHeap and replace with UseContainerSupport(see https://www.atamanroman.dev/articles/jvm-memory-settings-container-environment/)

This addresses issue #1

### CODE REVIEW

**Comment by brandonmcclure** [comment]
I am not certain of the ramifications of building older spigot releases with the newer java version, This allows it to build a 1.17 and 1.16.5 jar without any immediate issues. (IE no errors from build tools)

**Comment by sindastra** [comment]
> I am not certain of the ramifications of building older spigot releases with the newer java version

This was the first thing I was concerned about.

If 1.17 **and** 1.16.5 can be built, that's a good start.

However, as far as I know, 1.12.x seems to still be a popular version (I think due to the large variety of mods available).

Would you mind doing a test build of 1.12.x? While I don't think we necessarily should support 1.12.x as it's very old, it might be good to know, and perhaps I could create a legacy docker image for those who want to build older versions in case this update would break 1.12.x.

File: hub/build.sh [hunk 1]
```diff
@@ -6,4 +6,4 @@ rm -rf BuildTools.jar
 wget https://hub.spigotmc.org/jenkins/job/BuildTools/lastSuccessfulBuild/artifact/target/BuildTools.jar -O BuildTools.jar
 
 export MAVEN_OPTS="-Xmx2G"
-java -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -Xmx2G -jar BuildTools.jar "$@"
+java -XX:+UnlockExperimentalVMOptions -XX:+UseContainerSupport -Xmx2G -jar BuildTools.jar "$@"
```
Comment (sindastra-docker) [inline hub/build.sh:9]: ```UseContainerSupport``` should be enabled by default in Java 10+

**Comment by sindastra-docker** [comment]
OK, I have tested all versions from 1.12.2 - 1.17 and updating to Java 16 breaks building anything older than 1.16.5.

I am looking into a solution.

**Comment by sindastra-docker** [comment]
So, I looked into it, did a lot of testing, did preparations and in theory I could merge now.

However, there's a formality to be dealt with: Licensing.

Since I didn't expect anyone to contribute, I never chose a FOSS license. 

Would you be OK with me using something like the BSD, LGPL, GPL or AGPL license, and then merge your changes under that same license?

I know it might sound stupid, but it's one of those formalities that just have to be dealt with.

**Comment by brandonmcclure** [comment]
Awesome! I think the multiple tags is a great way to go with this. 
As for the licensing I totally understand, and am comfortable with any of those.

**Pull request merged**